File: models/course.dart
class Course {
  final String id;
  final String title;
  final String subject;
  final String teacherName;
  final String imageUrl;
  final double progress;
  final int totalLessons;
  final int completedLessons;
  final int? studentsCount;

  Course({
    required this.id,
    required this.title,
    required this.subject,
    required this.teacherName,
    required this.imageUrl,
    required this.progress,
    required this.totalLessons,
    required this.completedLessons,
    this.studentsCount,
  });
}

File: edubridge-backend/models/Course.js
// filepath: edubridge-backend/models/Course.js
const mongoose = require('mongoose');

const LessonSchema = new mongoose.Schema({
  title: { type: String, required: true },
  content: { type: String, required: true },
  videoUrl: { type: String },
  resources: [{ type: String }],
  quizzes: [{
    question: { type: String, required: true },
    options: [{ type: String, required: true }],
    correctAnswer: { type: Number, required: true },
  }],
});

const CourseSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String, required: true },
  instructor: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  imageUrl: { type: String },
  category: { type: String, required: true },
  level: { type: String, enum: ['beginner', 'intermediate', 'advanced'], required: true },
  price: { type: Number, default: 0 },
  duration: { type: Number, required: true }, // in hours
  lessons: [LessonSchema],
  enrolledStudents: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
  ratings: [{
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    rating: { type: Number, min: 1, max: 5 },
    review: { type: String },
    date: { type: Date, default: Date.now }
  }],
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Course', CourseSchema);